tests/devices/test_default_qubit.py

```diff
@@ -2292,43 +2309,28 @@ def test_s_inverse():
         expected = np.array([1.0, -1.0j]) / np.sqrt(2)
         assert np.allclose(dev.state, expected, atol=tol, rtol=0)
 
-    def test_inverse_S_decomposition(self, tol, mocker, monkeypatch):
-        """Test that applying the inverse of the S gate
-        works when the inverse S gate is decomposed"""
+    def test_inverse_decomposition(self, tol, mocker):
+        """Test that the expansion is inverted when inverse gate is applied and the gate isn't supported."""
```

Comment:
I think one reason we were testing the S gate specifically is that it was an edge case, the inverse wasn't defined in PL aside from `S.inv()`. This was an edge case that was quite buggy, and without this specific test it took a long time to find the bug.

I'm wondering if this test still covers that edge case?

Reply:
Can you give me more information about the problem? Then I can better rewrite the test.

Reply:
@antalszava, did you originally find/solve this problem/write this test? I can't remember 🤔

Reply:
Nope, only witnessed the resolution here :wink: It looks like the decomposition of S did not consider inversion and hence resulted in an incorrect list of operations.

* Original issue: https://github.com/PennyLaneAI/pennylane/issues/955
* PR: https://github.com/PennyLaneAI/pennylane/pull/956

Reply:
If I understand this issue correctly, then this modified test checks the exact same logic.  We need to make sure that the expansion takes into account in-place inversion.

Reply:
Nice! thanks for checking this